### DISCUSSION TITLE
Pdf table headers styling

### DISCUSSION THREAD

**Discussion by cowleyb** [py-pdf/fpdf2#924] [Q&A] [answered]
Hey I'm trying to use the .table to create a vertical table with labels in the first column and data in the second column. Im using .table instead of .multi_cell because I find it easier for rows to be automatically resized when text is wrapped onto a new line.  I dont want there to be any horizontal line in the first row. Previously in version 2.7.4 I could add a table.row() with no cells to make it think the first row as a header was already there which caused it to get rid of the horizontal lines in the first row.
Version 2.7.4:

<img width="446" alt="image" src="https://github.com/py-pdf/fpdf2/assets/60425597/bad990f0-78e8-42d8-aa8c-88332388288e">

<img width="826" alt="image" src="https://github.com/py-pdf/fpdf2/assets/60425597/a31f4b31-9d99-41ff-b857-f0ed10af7e9b">

However updating the version to the latest adds this if check, that causes my previous code to raise the exception. I have to remove the table.row()
<img width="586" alt="image" src="https://github.com/py-pdf/fpdf2/assets/60425597/6b20c818-fa6d-427f-bee1-bdd94a0b8cce">

Im wondering is there anyway in the latest version to get the table styled so that it also treats the border styling so no horizontal lines as if the first row isn't a header?
Version 2.7.5:
<img width="391" alt="image" src="https://github.com/py-pdf/fpdf2/assets/60425597/4d4e6bab-c11a-4800-92c1-ef4b7fc21028">

<img width="829" alt="image" src="https://github.com/py-pdf/fpdf2/assets/60425597/911d3cf1-d37f-44a1-b0db-c60b2fb9cab8">
(Horizontal line appears after first row)

Maybe this is the wrong approach and you also have a better idea?

**Comment by gmischler** [comment]
Welcome to fpdf2, @cowleyb.

<s>If you check the [table documentation](https://py-pdf.github.io/fpdf2/Tables.html#disable-table-headings), you will find that you can disable the generation of a header by using the argument `first_row_as_headings=False`. This should solve your problem.</s>
edited:
I just saw that you're actually already using that.
This would make the addition of that horizontal line a bug, which was apparently added in release 2.7.5.
That will obviously have to be addressed. Thanks for reporting.
Actually, there's a chance that #912 (not released yet) may also have fixed this problem, but that needs to be verified.

**Reply by Lucas-C** [reply]
@cowleyb Could you check if the latest `fpdf2` from this GitHub repo solves your problem, please?
You can [install it](https://py-pdf.github.io/fpdf2/#installation) this way:

    pip install git+https://github.com/py-pdf/fpdf2.git@master

**Reply by cowleyb** [reply]
Thanks for the replies, i've pulled the latest you suggested and still get the same format on the table:
<img width="758" alt="image" src="https://github.com/py-pdf/fpdf2/assets/60425597/d586c3a5-aff5-42ae-99a2-205036bd171c">
If you think its a bug im happy to raise it as an issue and try to fix it

**Reply by Lucas-C** [reply, marked as answer]
I opened #927 to try to fux this

Coudl you please review it @cowleyb? 🙂
